addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -827,7 +850,8 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
             this.fields = {};
             for (const [fieldName, field] of Object.entries(fields)) {
                 field.name = fieldName;
-                field.domain = field.domain || [];
+                const formDomain = _getDomain(this.formEl, field.name, field.type, field.relation);
+                field.domain = formDomain || field.domain || [];
```

Comment:
I think that `getDomain` should be called only if `field.domain` doesn't exist, so we avoid systematically looking for the field to be more efficient :) 
Something like
```js
field.domain = field.domain || _getDomain(this.formEl, field.name, field.type, field.relation) || []
```

Reply:
No, the idea of my fix is to consider the domain set in `FormEditorRegistry.fields` higher priority than the one returned by the backend

Reply:
Okok I thought the issue was just that it didn't find the domain because it was stored somewhere else but if you say that the domains can also be different then it's alright :wink:

Reply:
But maybe I could combine the field.domain and the _getDomain... :thinking: I will let the reviewer choose :)